### DISCUSSION TITLE
Build a string from multiples inputs

### DISCUSSION THREAD

**Discussion by JoeyNovais** [alpinejs/alpine#3311] [1. Help]
Hello there,

I'm kinda stuck when trying to build an SQL expression builder. Basically it looks like this:
![image](https://user-images.githubusercontent.com/2325059/205325368-e688884c-6d1a-43ed-bd8f-082bf0eb39ed.png)
So I'm having a lot of nested x-data and I'm struggling to build the SQL string that should result out of it. I don't find a way to look at my data and to redeem the values because they are in differents x-data.

Here's the data nested tree:

- sqlQuery
- groups (control the dropdown 'logic')
- conditions (control the number of lines I add or delete)
- attributes (control the three dropdown per row)

What I would like to do is to watch the dropdown and input and trigger a global function that look at the full data tree and build the SQL string out of it.

Any ideas of how to do that ? Or am I going the wrong way with the tool ?

**Comment by ekwoka** [comment]
This should mainly just be one component.

For the ore logic.

Have a few smaller utility components where need be that provide modelables to the parent component that builds the query.
